lib/train-winrm/connection.rb

```diff
@@ -107,15 +107,29 @@ def file_via_connection(path)
         Train::File::Remote::Windows.new(self, path)
       end
 
-      def run_command_via_connection(command, &data_handler)
+      def run_command_via_connection(command, opts = {}, &data_handler)
         return if command.nil?
 
         logger.debug("[WinRM] #{self} (#{command})")
         out = ""
+        response = nil
```

Comment:
Was there a bug that requires setting the `response` local variable earlier in the run? My gut tells me line 119 should be enough but I might be missing something

Reply:
119 is inside a block

```
jstocks@phobos train % bundle exec ruby winrm-test.rb
Traceback (most recent call last):
	4: from winrm-test.rb:14:in `<main>'
	3: from winrm-test.rb:14:in `times'
	2: from winrm-test.rb:14:in `block in <main>'
	1: from /Users/jstocks/code/train/lib/train/plugins/base_connection.rb:148:in `run_command'
/Users/jstocks/code/train-winrm/lib/train-winrm/connection.rb:135:in `run_command_via_connection': undefined local variable or method `response' for TrainPlugins::WinRM::Connection[unknown]:TrainPlugins::WinRM::Connection (NameError)
Did you mean?  respond_to?
```